backend/src/main/routes/AccountRoute.ts

```diff
@@ -135,19 +147,24 @@ export default class AccountRoute extends CommonRoutesConfig {
 
     this.getApp()
       .route(`/accounts/business`)
-      .post(async (req: express.Request, res: express.Response, next: express.NextFunction) => {
-        try {
-          const newBusinessAccount = await this.businessAccountService.createBusinessAccount(req.body);
-          const dto = JSON.parse(JSON.stringify(newBusinessAccount));
-          delete dto.account.password;
-          res.status(StatusCodes.CREATED).send(dto);
-        } catch (err) {
-          next(err);
+      .post(
+        checkJwt,
+        checkRole(new Set([Roles.ADMIN])),
+        async (req: express.Request, res: express.Response, next: express.NextFunction) => {
+          try {
+            const newBusinessAccount = await this.businessAccountService.createBusinessAccount(req.body);
+            const dto = JSON.parse(JSON.stringify(newBusinessAccount));
+            delete dto.account.password;
+            res.status(StatusCodes.CREATED).send(dto);
+          } catch (err) {
+            next(err);
+          }
         }
-      });
+      );
 
     this.getApp()
       .route(`/accounts/business/:email`)
+      .all(checkJwt, checkRole(new Set([Roles.ADMIN])))
```

Comment:
Although the role specified isn't wrong. I'm just wondering how we would handle the case where the business itself need to retrieve information about his business. Because this is probably the main use case of this API endpoint. 

We should re-evaluate how we want to solve this issue. One way is to allow business account to access this endpoint but only return data if the email passed matches the current user. You can obtain the current user's information through the jwt token.

Reply:
Fixed. Added a verification with the user email